src/SME.SGP.Aplicacao/Commands/PendenciaAusenciaFechamento/ExecutarVerificacaoPendenciaAusenciaFechamento/ExecutarVerificacaoPendenciaAusenciaFechamentoCommandHandler.cs

```diff
@@ -0,0 +1,157 @@
+﻿using MediatR;
+using Sentry;
+using SME.SGP.Aplicacao.Integracoes;
+using SME.SGP.Dominio;
+using SME.SGP.Infra;
+using SME.SGP.Infra.Dtos;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SME.SGP.Aplicacao
+{
+    public class ExecutarVerificacaoPendenciaAusenciaFechamentoCommandHandler : IRequestHandler<ExecutarVerificacaoPendenciaAusenciaFechamentoCommand, bool>
+    {
+        private readonly IMediator mediator;
+        private readonly IServicoEol servicoEol;
+        public ExecutarVerificacaoPendenciaAusenciaFechamentoCommandHandler(IMediator mediator, IServicoEol servicoEol)
+        {
+            this.mediator = mediator ?? throw new ArgumentNullException(nameof(mediator));
+            this.servicoEol = servicoEol ?? throw new ArgumentNullException(nameof(servicoEol));
+
+        }
+
+        public async Task<bool> Handle(ExecutarVerificacaoPendenciaAusenciaFechamentoCommand request, CancellationToken cancellationToken)
+        {
+            var turmas = await mediator.Send(new ObterTurmasPorAnoModalidadeQuery(DateTime.Now.Year, (int)request.Modalidade));
+
+            var periodosEscolares = await mediator.Send(new ObterPeriodosEscolaresPorModalidadeDataFechamentoQuery((int)request.ModalidadeTipoCalendario, DateTime.Now.AddDays(request.DiasParaGeracaoDePendencia)));
+            var componentes = await mediator.Send(new ObterComponentesCurricularesQuery());
+            foreach (var turma in turmas)
+            {
+                var ue = await mediator.Send(new ObterUEPorTurmaCodigoQuery(turma.CodigoTurma));
+                foreach (var periodoEscolar in periodosEscolares)
+                {
+                    var professoresTurma = await servicoEol.ObterProfessoresTitularesDisciplinas(turma.CodigoTurma);
+                    foreach (var professorTurma in professoresTurma)
+                    {
+                        var obterComponenteCurricular = componentes.FirstOrDefault(c => long.Parse(c.Codigo) == professorTurma.DisciplinaId);
+                        if (obterComponenteCurricular != null)
+                        {
+                            var verificaFechamento = await mediator.Send(new VerificaFechamentoTurmaDisciplinaPorIdCCEPeriodoEscolarQuery(turma.Id, long.Parse(obterComponenteCurricular.Codigo), periodoEscolar.Id));
```

Comment:
quando o bimestre for 2 ou 4 tem que fazer a mesma checagem com periodoEscolarId = null pra verificar se existe também fechamento final